distributed/client.py

```diff
@@ -486,13 +488,29 @@ def __init__(self, address=None, loop=None, timeout=5,
         super(Client, self).__init__(connection_args=self.connection_args,
                                      io_loop=self.loop)
 
-        self.start(timeout=timeout, asynchronous=asynchronous)
+        self.start(timeout=timeout)
 
-        from distributed.channels import ChannelClient
-        ChannelClient(self)  # registers itself on construction
         from distributed.recreate_exceptions import ReplayExceptionClient
         ReplayExceptionClient(self)
 
+    @property
+    def asynchronous(self):
+        """ Are we running in the event loop?
+
+        This is true if the user signaled that we might be when creating
+
+            Client(asynchronous=True)
+
+        and if we are currently in the same thread as the event loop.
+        """
+        result = self._asynchronous
+        try:
+            if get_thread_identity() != self.loop._thread_ident:
+                result = False
+        except AttributeError:  # AsyncIOLoop doesn't have _thread_ident
```

Comment:
It means that we just go with what the user provided when they defined the client.  We don't overrule them when we know that the situation has changed.